src/downshift.js

```diff
@@ -154,6 +155,7 @@ class Downshift extends Component {
     highlightedIndex = this.props.defaultHighlightedIndex,
     otherStateToSet = {},
   ) => {
+    otherStateToSet = pickState(otherStateToSet)
```

Comment:
Actually, I think this is the only place we really need this function called because all the other code paths lead here. Try removing the other changes and see if this works. I think it will.

Reply:
I just tried this, and it didn't work for me. I also tried to move this further down into `internalSetState`, and that didn't remove the warnings either.